src/graphics/camera_normal.cpp

```diff
@@ -344,17 +354,31 @@ void CameraNormal::positionCamera(float dt, float above_kart, float cam_angle,
 
     float tan_up = tanf(cam_angle);
 
-    if (getMode() == CM_SPECTATOR_SOCCER)
+    switch(getMode())
     {
-        SoccerWorld *soccer_world = dynamic_cast<SoccerWorld*> (World::getWorld());
-        if (soccer_world)
+    case CM_SPECTATOR_SOCCER:
+        {
+            SoccerWorld *soccer_world = dynamic_cast<SoccerWorld*> (World::getWorld());
+            if (soccer_world)
+            {
+                Vec3 ball_pos = soccer_world->getBallPosition();
+                Vec3 to_target=(ball_pos-wanted_target);
+                wanted_position = wanted_target + Vec3(0,  fabsf(distance)*tan_up+above_kart, 0) + (to_target.normalize() * distance);
+                m_camera->setPosition(wanted_position.toIrrVector());
+                m_camera->setTarget(wanted_target.toIrrVector());
+                return;
+            }
+            break;
+        }
+    case CM_SPECTATOR_TOP_VIEW:
         {
-            Vec3 ball_pos = soccer_world->getBallPosition();
-            Vec3 to_target=(ball_pos-wanted_target);
-            wanted_position = wanted_target + Vec3(0,  fabsf(distance)*tan_up+above_kart, 0) + (to_target.normalize() * distance);
+            SoccerWorld *soccer_world = dynamic_cast<SoccerWorld*> (World::getWorld());
+            if (soccer_world) wanted_target = soccer_world->getBallPosition();
+            wanted_position = wanted_target + Vec3(0,  distance+above_kart, 0);
             m_camera->setPosition(wanted_position.toIrrVector());
             m_camera->setTarget(wanted_target.toIrrVector());
             return;
+            break;
```

Comment:
![](https://cdn.lihkg.com/assets/faces/dog/math.gif)

Reply:
One can never be too careful.

Reply:
Is it a good think for you that i force push ? or do you prefer standard commits ?

Reply:
Tiny change better force push, btw can you add { } around the indentation change? ![](https://cdn.lihkg.com/assets/faces/mouse/please.gif)

Reply:
https://github.com/supertuxkart/stk-code/pull/4267/files#diff-ed59f4519babb9a8eb34ecd596442e8b

Reply:
🐍![](https://cdn.lihkg.com/assets/faces/pig/knife.gif)
![](https://na.cx/i/6NTmohU.jpg)